src/backend/storeManagers/hyperplay/games.ts

```diff
@@ -1475,3 +1564,387 @@ function writeManifestFile(
+          runner: 'hyperplay',
+          folder: gameInfo.install.install_path,
+          progress: {
+            folder: gameInfo.install.install_path,
+            totalSize,
+            percent,
+            diskSpeed: downloadSpeed / 1024 / 1024,
+            downSpeed: downloadSpeed / 1024 / 1024,
+            bytes: downloadedDataInMiB,
+            eta
+          }
+        })
+
+        logInfo(
+          `Progress: ${percent.toFixed(2)}%, Downloaded: ${getFileSize(
+            downloadedData
+          )}, Speed: ${getFileSize(
+            downloadSpeed
+          )}/s, ETA: ${eta} totalSize: ${totalSize} = ${getFileSize(
+            totalSize
+          )}`,
+          LogPrefix.HyperPlay
+        )
+      }
+    }
+    // need this to cover 100% of abort cases
+    if (aborted) {
+      rmSync(datastoreDir, { recursive: true })
+      return { status: 'abort' }
+    }
+
+    trackEvent({
+      event: 'Patching Success',
+      properties: {
+        game_name: gameInfo.app_name,
+        game_title: gameInfo.title,
+        platform: getPlatformName(platform),
+        platform_arch: platform
+      }
+    })
+
+    logInfo(`Patching ${appName} completed`, LogPrefix.HyperPlay)
+    rmSync(datastoreDir, { recursive: true })
+
+    return { status: 'done' }
+  } catch (error) {
+    logError(`Error while patching ${error}`, LogPrefix.HyperPlay)
+
+    trackEvent({
+      event: 'Patching Failed',
+      properties: {
+        error: `${error}`,
+        game_name: gameInfo.app_name,
+        game_title: gameInfo.title,
+        platform: getPlatformName(platform!),
+        platform_arch: platform!
+      }
+    })
+
+    captureException(new Error(`Error while patching ${error}`), {
+      extra: {
+        method: 'applyPatching',
+        appName,
+        title: gameInfo.title,
+        install_path,
+        platform,
+        version,
+        newVersion
+      }
+    })
+    rmSync(datastoreDir, { recursive: true })
+    return { status: 'error', error: `Error while patching ${error}` }
+  }
+}
+
+async function getManifest(
+  appName: string,
+  platformName: string,
+  version: string
+) {
+  try {
+    const manifestPath = path.join(
+      ipdtManifestsPath,
+      `${appName}-${platformName}-${version}.json`
+    )
+
+    if (!existsSync(manifestPath)) {
+      logDebug(
+        `Manifest for ${appName} not found for version ${version} and platform ${platformName}, downloading it.`,
+        LogPrefix.HyperPlay
+      )
+
+      await downloadGameIpdtManifest(appName, version)
+    }
+
+    return manifestPath
+  } catch (error) {
+    logError(
+      `Error in getManifest for ${appName}: ${error}`,
+      LogPrefix.HyperPlay
+    )
+    captureException(
+      new Error(`Error in getManifest for ${appName}: ${error}`),
+      {
+        extra: {
+          method: 'getManifest',
+          appName,
+          platformName,
+          version
+        }
+      }
+    )
+    return ''
```

Comment:
what's the benefit of returning an empty string as the manifest?

Reply:
None, remove this line, it will never get there.